src/tools/android/java/com/google/devtools/build/android/ResourceProcessorBusyBox.java

```diff
@@ -183,43 +182,51 @@ private static int runPersistentWorker() throws Exception {
     PrintStream ps = new PrintStream(buf, true);
     PrintStream realStdOut = System.out;
     PrintStream realStdErr = System.err;
-    try {
-      // Redirect all stdout and stderr output for logging.
-      System.setOut(ps);
-      System.setErr(ps);
-
-      while (true) {
-        try {
-          WorkRequest request = WorkRequest.parseDelimitedFrom(System.in);
-          if (request == null) {
-            break;
-          }
-
-          int exitCode = processRequest(request.getArgumentsList());
-          ps.flush();
-
-          WorkResponse.newBuilder()
-              .setExitCode(exitCode)
-              .setRequestId(request.getRequestId())
-              .setOutput(buf.toString())
-              .build()
-              .writeDelimitedTo(realStdOut);
 
-          realStdOut.flush();
-          buf.reset();
-        } catch (IOException e) {
-          logger.severe(e.getMessage());
-          e.printStackTrace(realStdErr);
-          return 1;
-        }
-      }
+    // Redirect all stdout and stderr output for logging.
+    System.setOut(ps);
+    System.setErr(ps);
+    try {
+      WorkRequestHandler workerHandler =
+          new WorkRequestHandler.WorkRequestHandlerBuilder(
+                  new WorkRequestHandler.WorkRequestCallback(
+                      (request, pw) -> processRequest(request.getArgumentsList(), pw, buf)),
+                  realStdErr,
+                  new ProtoWorkerMessageProcessor(System.in, realStdOut))
+              .setCpuUsageBeforeGc(Duration.ofSeconds(10))
+              .build();
+      workerHandler.processRequests();
+    } catch (IOException e) {
+      logger.severe(e.getMessage());
+      e.printStackTrace(realStdErr);
+      return 1;
     } finally {
       System.setOut(realStdOut);
       System.setErr(realStdErr);
     }
     return 0;
   }
 
+  private static int processRequest(List<String> args, PrintWriter pw, ByteArrayOutputStream buf) {
+    int exitCode;
+    try {
+      // Process the actual request and grab the exit code
+      exitCode = processRequest(args);
+    } catch (Exception e) {
+      e.printStackTrace(pw);
+      exitCode = 1;
+    } finally {
+      // Write the captured buffer to the work response
+      String captured = buf.toString().trim();
+      buf.reset();
+      if (!captured.isEmpty()) {
+        pw.print(captured);
```

Comment:
This preserves the previous logging behavior seen here https://github.com/bazelbuild/bazel/pull/14428/files#diff-6bbb0aaf4cf2d425734021815617e2b6203223abe7dc76ca8030a90e9f043873L204